### PR TITLE
added deinit and visual tests

### PR DESCRIPTION
**Pull request by acaldas**
https://github.com/ripe-tech/ripe-core/issues/3444

### CODE REVIEW

**Comment by joamag** [comment]
@acaldas need to discuss this over the phone

File: src/python/ripe_demo/static/js/ripe.js [hunk 1]
```diff
@@ -2092,6 +2146,26 @@ ripe.Configurator.prototype._registerHandlers = function() {
...
 
+ripe.Configurator.prototype._addElementHandler = function(event, callback) {
+    this.element.addEventListener(event, callback);
+
+    var callbacks = this.elementEvents[event] || [];
+    callbacks.push(callback);
+    this.elementEvents[event] = callbacks;
+};
+
+ripe.Configurator.prototype._removeElementHandlers = function() {
```
Comment (acaldas) [inline src/python/ripe_demo/static/js/ripe.js]: comment this function

File: src/python/ripe_demo/static/js/ripe.js [hunk 2]
```diff
@@ -2234,15 +2323,22 @@ ripe.Image.prototype.setInitialsBuilder = function(builder, options) {
...
-    this.element.addEventListener("load", function() {
+    this.loadListener = function() {
         this.trigger("loaded");
-    }.bind(this));
+    }.bind(this);
+    this.element.addEventListener("load", this.loadListener);
+
     var Observer = MutationObserver || WebKitMutationObserver; // eslint-disable-line no-undef
-    var observer = Observer ? new Observer(function(mutations) {
+    this.observer = Observer ? new Observer(function(mutations) {
```
Comment (acaldas) [inline src/python/ripe_demo/static/js/ripe.js]: this._observer

File: src/js/visual/configurator.js [hunk 3]
```diff
@@ -876,6 +891,26 @@ ripe.Configurator.prototype._registerHandlers = function() {
     ripe.touchHandler(this.element);
 };
 
+ripe.Configurator.prototype._addElementHandler = function(event, callback) {
```
Comment (acaldas) [inline src/js/visual/configurator.js]: move to Visual

**Pull request merged**